addons/mail/static/src/core/common/partner_suggestion_compare_fn.js

```diff
@@ -0,0 +1,141 @@
+ * Note that the compare functions will be call in the context of the
+ * caller (i.e. the SuggestionService).
+ */
+export const partnerSuggestionCompareRegistry = registry.category(
+    "mail.partner_suggestion_compare_fn"
+);
+
+// Ordering:
+// - recent chat partners
+// - internal users
+// - channel members
+// - thread followers
+// - longgest match of name, alphabetically if having same length of match
+// - longgest match of email, alphabetically if having same length of match
+// - id
+//
+
+partnerSuggestionCompareRegistry.add("mail.recent-chats", {
+    sequence: 0,
+    fn(p1, p2) {
+        const recentChatPartnerIds = this.personaService.getRecentChatPartnerIds();
+        const recentChatIndex_p1 = recentChatPartnerIds.findIndex(
+            (partnerId) => partnerId === p1.id
+        );
+        const recentChatIndex_p2 = recentChatPartnerIds.findIndex(
+            (partnerId) => partnerId === p2.id
+        );
+        if (recentChatIndex_p1 !== -1 && recentChatIndex_p2 === -1) {
+            return -1;
+        } else if (recentChatIndex_p1 === -1 && recentChatIndex_p2 !== -1) {
+            return 1;
+        } else if (recentChatIndex_p1 < recentChatIndex_p2) {
+            return -1;
+        } else if (recentChatIndex_p1 > recentChatIndex_p2) {
+            return 1;
+        }
+    },
+});
+
+partnerSuggestionCompareRegistry.add("mail.internal-users", {
+    sequence: 10,
+    fn(p1, p2) {
+        const isAInternalUser = p1.user?.isInternalUser;
+        const isBInternalUser = p2.user?.isInternalUser;
+        if (isAInternalUser && !isBInternalUser) {
+            return -1;
+        }
+        if (!isAInternalUser && isBInternalUser) {
+            return 1;
+        }
+    },
+});
+
+partnerSuggestionCompareRegistry.add("mail.members", {
+    sequence: 15,
+    fn(p1, p2, { thread } = {}) {
+        if (thread?.model === "discuss.channel") {
+            const isMember1 = thread.channelMembers.some((member) => member.persona === p1);
+            const isMember2 = thread.channelMembers.some((member) => member.persona === p2);
+            if (isMember1 && !isMember2) {
+                return -1;
+            }
+            if (!isMember1 && isMember2) {
+                return 1;
+            }
+        }
+    },
+});
+
+partnerSuggestionCompareRegistry.add("mail.followers", {
+    sequence: 20,
+    fn(p1, p2, { thread } = {}) {
+        if (thread) {
+            const followerList = [...thread.followers];
+            const isFollower1 = followerList.some((follower) => follower.partner === p1);
+            const isFollower2 = followerList.some((follower) => follower.partner === p2);
+            if (isFollower1 && !isFollower2) {
+                return -1;
+            }
+            if (!isFollower1 && isFollower2) {
+                return 1;
+            }
+        }
+    },
+});
+
+partnerSuggestionCompareRegistry.add("mail.name", {
+    sequence: 25,
+    fn(p1, p2, { searchTerms } = {}) {
```

Comment:
searchTerms can be singular, and no need for default object as it is always given